web/tests/test_chapters.py

```
@@ -0,0 +1,629 @@
+            title="Test Resource",
+            description="A test resource",
+            resource_type="document",
+            file=test_file,
+            created_by=self.user,
+        )
+
+        assert resource.chapter == self.chapter
+        assert resource.title == "Test Resource"
+        assert resource.description == "A test resource"
+        assert resource.resource_type == "document"
+        # File‑based resources should *not* have an external URL
```

Ruff (0.8.2)

Comment contains ambiguous `‑` (NON-BREAKING HYPHEN). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)